From 28e9631e86a47d02ca23ce1ca83700445d88a6ca Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Fri, 9 Jan 2015 00:27:25 -0500
Subject: [iscsi-target] Allow ->MaxXmitDataSegmentLength assignment for iser discovery

Message-id: <1420763250-27236-5-git-send-email-agrover@redhat.com>
Patchwork-id: 102559
O-Subject: [RHEL7.1 PATCH e-stor 4/9] iscsi-target: Allow ->MaxXmitDataSegmentLength assignment for iser discovery
Bugzilla: 1058736
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Nicholas Bellinger <nab@linux-iscsi.org>

This patch changes iscsi_set_connection_parameters() to allow
conn_ops->MaxXmitDataSegmentLength assignement to occur during
in-band iser send-targets discovery, as this value is required
by TEXT response processing code.

Reported-by: Or Gerlitz <ogerlitz@mellanox.com>
Cc: Or Gerlitz <ogerlitz@mellanox.com>
Cc: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/target/iscsi/iscsi_target_parameters.c b/drivers/target/iscsi/iscsi_target_parameters.c
index e382221..35fd643 100644
--- a/drivers/target/iscsi/iscsi_target_parameters.c
+++ b/drivers/target/iscsi/iscsi_target_parameters.c
@@ -1799,9 +1799,6 @@ void iscsi_set_connection_parameters(
    * this key is not sent over the wire.
    */
   if (!strcmp(param->name, MAXXMITDATASEGMENTLENGTH)) {
-   if (param_list->iser == true)
-    continue;
-
    ops->MaxXmitDataSegmentLength =
     simple_strtoul(param->value, &tmpptr, 0);
    pr_debug("MaxXmitDataSegmentLength:     %s\n",
-- 
1.7.1